Comseg cli

tdefa opened the pull request:
implementation of ComSeg cli 

- An example to run comseg in CLI is in  sopa/docs/tutorials/cli_other_segmentation.md
- documentation at "sopa/docs/cli.md" was update fro comseg 

A function transcript_segmentation() was created to handle both Baysor and ComSeg

quentinblampey reviewed (commented):
Thanks, looking good! I added a few comments

quentinblampey commented:
Here is a todo list for the next steps:
- [ ] Update the Snakefile and utils.py
- [ ] Have a complete ComSeg tutorial for API + CLI + snakemake
- [ ] Have a ComSeg config for the toy dataset (used in the above snakemake tutorial)
- [ ] Have a ComSeg config for a few major technologies (e.g., MERSCOPE and Xenium)

tdefa commented on docs/tutorials/cli_other_segmentation.md:
```diff
@@ -0,0 +1,76 @@
+
+### Option 3: ComSeg
```
Ok I can gather other segmentation and cli_other_segmentation in the next pr

tdefa commented on sopa/cli/segmentation.py:
```diff
@@ -176,3 +181,46 @@ def _run_staining_segmentation(
+    sdata_path: str = typer.Argument(help=SDATA_HELPER),
+    patch_index: int = typer.Option(
+        default=None,
+        help="Index of the patch on which the segmentation method should be run. NB: the number of patches is `len(sdata['sopa_patches'])`",
```
ok

tdefa commented on sopa/cli/segmentation.py:
```diff
@@ -176,3 +181,46 @@ def _run_staining_segmentation(
+    ),
+    patch_dir: str = typer.Option(
+        default=None,
+        help="Path to the temporary the segmentation method directory inside which we will store each individual patch segmentation. By default, saves into the `.sopa_cache/<method_name>` directory",
```
ok

tdefa commented on sopa/cli/segmentation.py:
```diff
@@ -1,13 +1,18 @@
 
 import ast
+import logging
+from pathlib import Path
```
Ok I moved it inside

tdefa commented on sopa/cli/patchify.py:
```diff
@@ -70,7 +72,108 @@ def baysor(
+
+
+@app_patchify.command()
+def transcript_segmentation(
```
ok

tdefa commented on sopa/cli/patchify.py:
```diff
@@ -70,7 +72,108 @@ def baysor(
+@app_patchify.command()
+def transcript_segmentation(
+    sdata_path: str = typer.Argument(help=SDATA_HELPER),
+    method: str = typer.Option(
```
Oks

tdefa replied:
ok

quentinblampey commented:
Thanks for the changes! Regarding the CI, you can install `pre-commit` locally to fix the issues:
```sh
pre-commit install

pre-commit run --all-files
```

tdefa commented:
> Thanks for the changes! Regarding the CI, you can install `pre-commit` locally to fix the issues:
> 
> ```shell
> pre-commit install
> 
> pre-commit run --all-files
> ```

Ok thanks I just saw it

tdefa commented:
I added comseg to the snakemake pipeline + few config file examples.

Now I will complete ComSeg tutorial for API + CLI + snakemake all in the file  : "other_segmentations.ipynb" 
do you agree ?

quentinblampey commented:
Thanks for your updates, it looks very good!

I fixed and cleaned a few things

Also, I had some issues running comseg, now I get an error `'PCA' object has no attribute 'explained_variance_'` when running the patch step of comseg, do you know why?
Two other things:
- Installing comseg is not enough to have all the dependencies, I also needed to install alphashape / igraph / leidenalg. Do you think these could be added as a dependency of ComSeg for convenience?
- I moved the tutorial in `docs/tutorial/comseg.ipynb`, can you check and fill the "Snakemake" and "CLI" sections? It can be pretty short, since it is very similar to the existing tutorial. For instance, for the Snakemake pipeline, you can provide the command line to run the toy dataset, and for the CLI you can do something similar to the CLI tutorial with Baysor.
- Why do you need to do `importlib.reload(method)`?

After this, I will merge the PR!

tdefa commented:
Ok thanks  reviewing! 


- `PCA' object has no attribute 'explained_variance_'` I will look at it and also complete dependency of ComSeg


- for` importlib.reload(method)` I will remove it I think it is a mistake

I will do additional tests on other data this week, before the next push.

tdefa commented:
Hello,

Sorry it took so long, I think that it is now ready for your review.

I added `min_transcripts_per_patch` and `min_cells_per_patch options` to the CLI for ComSeg. Additionally, I included support for  other segmentation than Cellpose for prior initialization, which is useful for protocols that already segment nuclei, such as Xenium. I also updated the Snakefile with checkpoint patchify_comseg. An example config for Snakemake + Xenium + ComSeg has been added.

I modified the valid index patches of ComSeg so it run only on crops with nuclei.

I updated the ComSeg dependency in the new pip version. However, I did not encounter the `PCA object has no attribute 'explained_variance_' issue, so I do not know why it occurred for you.

quentinblampey reviewed (approved):
Looking good to me, thanks @tdefa!
